test/strategies/elkilpstrategy.js

```diff
@@ -0,0 +1,316 @@
+            elkTokenAddress, // Address of swapPair token 0 (ELK)
+            timelockAddress, // Timelock contract for YY
+            1,
+            1,
+            1,
+            1
+        )
+        await elkIlpStrategyV5.deployed()
+
+        wavaxTokenContract = await ethers.getContractAt("IWAVAX", wavaxTokenAddress)
+        elkTokenContract = await ethers.getContractAt("IElkERC20", elkTokenAddress)
+        //makes sure owner has enough WAVAX balance
+        if ((await wavaxTokenContract.balanceOf(owner.address)).lt("1000000000000000000000")) {
+            wavaxTokenContract.connect(owner).deposit({
+                value: BigNumber.from("1000000000000000000000")
+                    .sub(await wavaxTokenContract.balanceOf(owner.address))
+            })
+        }
+        if ((await wavaxTokenContract.balanceOf(account1.address)).lt("1000000000000000000000")) {
+            wavaxTokenContract.connect(account1).deposit({
+                value: BigNumber.from("1000000000000000000000")
+                    .sub(await wavaxTokenContract.balanceOf(account1.address))
+            })
+        }
+    })
+
+    describe("Test basic functionalities", async () => {
+
+    it('Can deploy', async () => {
+        expect(await elkIlpStrategyV5.owner()).to.equal(timelockAddress)
+    })
+
+    it('Set correctly the _minTokensToReinvest', async () => {
+        expect(await elkIlpStrategyV5.MIN_TOKENS_TO_REINVEST()).to.equal(1)
+    })
+
+    it('Set correctly the _adminFeeBips', async () => {
+        expect(await elkIlpStrategyV5.ADMIN_FEE_BIPS()).to.equal(1)
+    })
+
+    it('Set correctly the _devFeeBips', async () => {
+        expect(await elkIlpStrategyV5.DEV_FEE_BIPS()).to.equal(1)
+    })
+
+    it('Set correctly the _reinvestRewardBips', async () => {
+        expect(await elkIlpStrategyV5.REINVEST_REWARD_BIPS()).to.equal(1)
+    })
+
+    it('Set correctly the devAddr', async () => {
+        expect(await elkIlpStrategyV5.devAddr()).to.equal(owner.address)
+    })
+
+    it('Set correctly the allowance on the deposit token', async () => {
+        expect(await elkPairContract.allowance(elkIlpStrategyV5.address, elpStakingRewardAddress)).to.equal(MAX_UINT)
+    })
+
+    });
+
+    describe("Test deposit functionality", async () => {
+
+        beforeEach(async () => {
+            // We approve addresses
+            await wavaxTokenContract.approve(elkRouterContract.address, BigNumber.from("1000000000000000000000000"));
+            await elkTokenContract.approve(elkRouterContract.address, BigNumber.from("1000000000000000000000000"));
+            await elkPairContract.approve(elkIlpStrategyV5.address, BigNumber.from("1000000000000000000000000"));
+
+            await elkRouterContract.swapExactTokensForTokens(BigNumber.from("10000000000000000"), BigNumber.from("5000000000000000"), [wavaxTokenContract.address, elkTokenContract.address], owner.address, 1807909162115)
```

Comment:
would it work to use BigNumber.from(1e16) instead? it feels easier to read, same applying to all other BigNumber occurrences